### DISCUSSION TITLE
Commit Message with LineFeed becomes garbled text

### DISCUSSION THREAD

**Discussion by BrunoJuchli** [gitextensions/gitextensions#10777] [Q&A] [answered]
When I create a commit in GitExtensions with a multiline message where lines are separated by line feeds (Shift+Enter) it all looks ok in GitExtensions, but in other places the line feed seems to be an unrecognized character.

Does anyone have an idea what this could be caused by and how I could fix this?

### Git Extensions
![grafik](https://user-images.githubusercontent.com/2090172/221567686-80b8b5fe-85a4-43ef-b562-6500b4d80d98.png)

### GitHub
![grafik](https://user-images.githubusercontent.com/2090172/221567641-c7c0d4cd-c025-4643-b7e8-08e889cd41df.png)

### Rider (JetBrains IntelliJ)
![grafik](https://user-images.githubusercontent.com/2090172/221568093-18aa9e57-646a-4c36-9c7b-c0ba3c1813ad.png)

In another view it shows the (supposed) new line character as an empty rectangle.

**Comment by vbjay** [comment]
Don't press shift enter. It's that simple. Character 11 vs character 10
mean different things.

**Reply by RussKie** [reply]
If you look closely, Rider renders the vertical tabs too, though in a better way than GitHub:
![image](https://user-images.githubusercontent.com/4403806/221738167-e595533d-c9c7-48c4-9773-88a7e297e460.png)

**Comment by pmiossec** [comment, marked as answer]
Already discussed in #10724 and #10409

>  how I could fix this?

Just use 'Enter' instead of 'Shift+Enter' (until someone spend some time on it to fix the default behavior). And if you want to fix already done commits, you will have to rewrite the history with all the impacts it could have on other team members. So you need to ponder if it's worth it...

**Comment by BrunoJuchli** [comment]
Thank you guys for taking the time to answer.
I tried it out and it works.. just a question of habit.

Like written by someone in one of the referenced issues I also used Shift+Enter just to make sure I don't commit.
The context here is that in some other applications (including web) an enter press leads to submitting the form/confirming something - and not to a new line. That's why I came to the habit of using Shift+Enter.

**Reply by vbjay** [reply]
There is no need to hit shift enter. Even if you accidentally commit, you can amend or reset back 1 commit soft and then commit again. In the textbox enter will never cause commit to run.

**Reply by BrunoJuchli** [reply]
@vbjay 
Yes I get that, it's something I use rather regularly in my work process.
The habit of using shift-enter was formed by using other tools, where using enter was problematic and shift+enter circumvented the problems.
So that leads me to _always_ using Shift+Enter, no matter the tool, because it was an universal safe way to get a new line.
Until I came across the exception of GitExtensions, which is the only tool I use where this doesn't apply...

But thank you for the comment, I think it might be valuable to other readers who might stumble upon this and now know they don't need to worry much.
